flaskr/backservice.py

```diff
@@ -0,0 +1,181 @@
+    except (psycopg2.Error) as error:             
+        return make_response(jsonify({"msg": f"{DB_NOT_WORKING_MSG} \n {error}"}), 400, CONTENT_HEADER)
+        #TODO: LOG THE ERROR
+    finally:
+        if(connection is not None):
+            connection.close()
+
+
+# Return all the records that match the args parameters
+def createQueryByRequestType(args: dict, request_type: str, action: str = None):
+    query = ""
+
+    # Process each set of values
+    record = helper.argsToSetOfValues(args)
+    
+    #TODO: THIS METHOD SEEMS VERY WASTEFULLY BECAUSE WE CREATE NEW STRING EACH TIME
+    if(request_type == "GET" or request_type == "DELETE"):
+
+        query = f"SELECT * FROM {TABLE_NAME} WHERE " if request_type == "GET" else f"DELETE FROM {TABLE_NAME} WHERE "
+        # SELECT * FROM TABLE_NAME WHERE key1=value1 and key2=value2 and ...
+        query += " AND ".join(["{}='{}'".format(key, value) for key, value in record.items()]) 
+
+        return query
+    
+    if(request_type == "POST"):
+        # INSERT INTO TABLE_NAME(id, first_name, last_name, email, job_id) VALUES (%s, %s, %s, %s, %s)
+        if action == "create":
+            values = []
+            columns = []
+            for key, value in record.items():
+                columns.append(key)
+                values.append(f"\'{value}\'")
+            
+            query = "INSERT INTO {} ({}) VALUES ({})".format(TABLE_NAME, ", ".join(columns), ", ".join(values))
+
+        if(action == "update"):
+            key_value_pairs = ", ".join(["{}='{}'".format(key, value) for key, value in record.items()])            
+            query = f"UPDATE {TABLE_NAME} SET {key_value_pairs} WHERE {KEY_COLUMNS_NAMES}='{record.get(KEY_COLUMNS_NAMES)}'"
+        
+        return query
+
+
+@app.route("/candidate", methods=["GET", "POST", "DELETE"])
+def apiHandelCandidate():
+    GET_ALL_CANDIDATES = f"SELECT * FROM {TABLE_NAME} "
+    # TODO: CHANGE THE 100 TO A VARIABLE FROM USERS WITH DEFAULT OF 100
+    
+    messages = {
+        "get": {
+            "success": "Successful in fetching all candidates",
+            "error": "Candidates data for GET request not found"
+        },
+        "create": {
+            "success": "Successful in creating all new candidates",
+            "error": "Candidates data for POST (create) request not found"
+        },
+        "update": {
+            "success": "Successful in updating all candidates",
+            "error": "Candidates data for POST (update) request not found"
+        },
+        "delete": {
+            "success": "Successful in deleting all candidates",
+            "error": "Candidates data for DELETE request not found"
+        }
+    }
+    query = ""
+    success = ""
+    error = ""
+
+    try:
+        # Store the url query params
+        args = request.args.to_dict() 
+        
+        # Check connectivity to DB
+        getReady = apiHandelReady()
```

Comment:
I understand what you are saying, although if I remove this I will need to add more exceptions for a precise error message - this will cause a duplication of code and if I will split this method then it will casue more duplication